utils/das_data_sorter.py

```diff
@@ -1,76 +1,170 @@
 import pandas as pd 
-import argparse 
-from math import ceil
+from argparse import ArgumentParser
+from numpy import ceil
+from typing import TypeVar
 
-# averages data within the seconds, according to seconds array 
-def avg_Data(array):
-    avg_array = []
+D = TypeVar('DAS_data') # data type to represent columns of the data in a given file
 
-    for index_array in new_time:
-        total = 0
-        for index in index_array:
-            total += array[index]
-
-        avg = round(total/len(index_array), ndigits=2)
-        avg_array.append(avg)
-    return avg_array
-
-# accepts terminal commands
-parser = argparse.ArgumentParser()
+# accepts terminal arguments
+parser = ArgumentParser()
 parser.add_argument("--file", help="Input CSV file", action="store", required=True)
 parser.add_argument("--output", help="Returns the filtered data", default="filtered_data.csv", action="store")
 parser.add_argument("--unit", help="Specify time units (seconds, s, or minutes, m)", default="seconds", 
                     choices=["seconds", "s", "minutes", "m"], action="store")
 args = parser.parse_args()
 
-# initialising data
-data = pd.read_csv(args.file)
-
-milliseconds = data["time"]
-gps_course = data["gps_course"]
-gps_speed = data["gps_speed"]
-tempC = data["thermoC"]
-tempF = data["thermoF"]
-cadence = data["cadence"]
-power = data["power"]
-reed_vel = data["reed_velocity"]
-reed_dis = data["reed_distance"]
-
-time = []
-if args.unit == "seconds" or args.unit == "s":
-    time = milliseconds/1000
-elif args.unit == "minutes" or args.unit =="m":
-    time = milliseconds/1000/60
-
-# sorts time into specific time unit
-new_time = []
-previous_time = 0
-for i in range(len(time)):
-    if time[i] > previous_time:
-        # sets new previous_time, pushes current millisecond_indices into seconds array, creates new millisecond_indices
-        # millisecond_indices is an array of indexes for milliseconds within the same specified unit
-        if previous_time != 0:
-            # ignores first iteration
-            new_time.append(millisecond_indices)
-
-        previous_time = ceil(time[i])
-        millisecond_indices = []
-
-    # pushes index into current millisecond_indices
-    millisecond_indices.append(i)
-
-new_time.append(millisecond_indices)
-
-# writes newly-manipulated data into new csv file
-final = pd.DataFrame({
-    "time": range(1, len(new_time)+1),
-    "gps_course": avg_Data(gps_course),
-    "gps_speed": avg_Data(gps_speed),
-    "tempC": avg_Data(tempC),
-    "tempF": avg_Data(tempF),
-    "cadence": avg_Data(cadence),
-    "power": avg_Data(power),
-    "reed_velocity": avg_Data(reed_vel),
-    "reed_distance": avg_Data(reed_dis)
-})
-final.to_csv(args.output, index=False)
+class DasSort:
+    def __init__(self, data:str) -> None:
+        self.indexes = None
+        self.time = self.convert_time(data["time"])
+        self.gps = self.average_data(data["gps"])
```

Comment:
As far as I know, `gps` contains either a 1 when gps data is available, and 0 otherwise. Rather than averaging this data, I think it would make most sense to have a 1 if there's any 1s in the original data, and only a 0 if all values are 0s.

Reply:
I understand what you are saying, and it would make the script run quicker if it didn't use the average_data method. One problem is that I'm not sure how to go about this. 

When I tried checking if there was a 1 within the time interval, by doing `if 1 in data[index_array]`, it didn't work the way it's supposed to work. After doing some testing on a different Python file, I found that the data points inside the gps column is **not** of the integer type
![image](https://user-images.githubusercontent.com/58575113/82115789-4c61d000-97a8-11ea-9856-49ccdf4127e8.png)

Reply:
That's odd, it seems to be working for me
![image](https://user-images.githubusercontent.com/17876556/82132286-8b803780-9821-11ea-94d3-74052297a165.png)